agent/rpc/peering/service.go

```diff
@@ -476,8 +473,13 @@ func (s *Server) validatePeeringLocality(token *structs.PeeringToken, partition
 
 	// If the token has the same server name as this cluster, but we can't find the peering
 	// in our store, it indicates a naming conflict.
-	if s.Backend.GetServerName() == token.ServerName && peering == nil {
-		return fmt.Errorf("conflict - peering token's server name matches the current cluster's server name, %q, but there is no record in the database", s.Backend.GetServerName())
+	serverName, _, err := s.Backend.GetTLSMaterials()
+	if err != nil {
+		return fmt.Errorf("failed to fetch TLS materials: %w", err)
+	}
+
+	if serverName != "" && token.ServerName != "" && serverName == token.ServerName && peering == nil {
```

Comment:
Actually this needs some more thinking regarding peering a server with Connect to one without or a server with TLS to one without.

Reply:
TLS is required when generating a token now, so the token.ServerName will always be populated.